### PR TITLE
pre-commit: Add initial config and run it for all files

### PR DESCRIPTION
**Pull request by xdustinface**
This PR adds a first version of `.pre-commit-config.yaml` and applies the changes required by the following initial hooks in separate commits: 

- Some [general hooks](https://github.com/xdustinface/chia-blockchain/blob/d1bd18fe00da926c61d942f165c5b50fac4dd635/.pre-commit-config.yaml#L5-L10), for their description see https://github.com/pre-commit/pre-commit-hooks#hooks-available
- `black`
- `flake8`
- `mypy`

To use this you need to install `pre-commit`, see https://pre-commit.com/#installation

Then you can either run it manually with `pre-commit run` (changed files only) `pre-commit run --all-files` (all files in the repo) or it can just be installed as actual pre-commit hook in git by `pre-commit install` which then forces a call of all the above hooks before each commit.

Let me know if there are any concerns about any of the hooks or suggestions for others.

### CODE REVIEW

File: chia/wallet/util/debug_spend_bundle.py [hunk 1]
```diff
@@ -65,7 +65,6 @@ def debug_spend_bundle(spend_bundle, agg_sig_additional_data=bytes([3] * 32)) ->
             print("*** BAD PUZZLE REVEAL")
             print(f"{puzzle_reveal.get_tree_hash().hex()} vs {coin_solution.coin.puzzle_hash.hex()}")
             print("*" * 80)
-            breakpoint()
```
Comment (hoffmang9) [inline chia/wallet/util/debug_spend_bundle.py:68]: Was this intentional?

Comment (xdustinface) [reply chia/wallet/util/debug_spend_bundle.py:68]: yeah that's the `debug-statements` hook which makes sure there are is no debug stuff commited

**Comment by hoffmang9** [comment]
You can ignore the installer build failure or rebase on main as it's fixed there.

**Comment by xdustinface** [comment]
> You can ignore the installer build failure or rebase on main as it's fixed there.

Im up with main, the problem here is that i usually open my PRs from `xdustinface/chia-blockchain` not from `Chia-Network/chia-blockchain` and there is still the issue with the credentials we talked about in the standup today https://github.com/Chia-Network/chia-blockchain/pull/7053/checks?check_run_id=2948164782#step:19:6

**Comment by xdustinface** [comment]
Ahh i see you were talking about #7043, hmm doesn't seems like that helped for the fork repo situation because its in my branch :thinking:

**Comment by hoffmang9** [comment]
I see the step that Niraj overlooked and just PR'ed it.

**Comment by hoffmang9** [comment]
That wasn't meant to close this. Odd.

**Comment by xdustinface** [comment]
Any idea why LGTM JS is failing here? :thinking:

**Comment by hoffmang9** [comment]
LGTM js is just confused about the merge history. Ignore it.

**Comment by aqk** [comment]
Looks good, let's merge it when Bill is ready.

**Review by duongxuantuan154 — APPROVED** [review]
- [ ]

**Review by hoffmang9 — APPROVED** [review]
+1

**Pull request merged**